### PR TITLE
Enable/Disable downloading of a file via Settings

### PR DESCRIPTION
**Pull request by jamez14**
This change adds a setting to allow for the disabling of downloading a file. By default, the setting is checked (so downloading files will be enabled).

This change follows the same structure as the other settings, saving them in the store and retrieving them from the same place, by name. On hover, when disabled, the icon does not change and clicking the icon does nothing.

This PR should also satisfy the issue #186

### CODE REVIEW

File: src/view.tree.js [hunk 1]
```diff
@@ -98,13 +106,18 @@ class TreeView {
...
 
     // handle icon click, fix #122
     if ($target.is('i.jstree-icon')) {
       $target = $target.parent()
-      download = true
+      if (enableDownloading) {
+        download = true
+      } else {
+        download = false
+      }
```
Comment (Ephemera) [inline src/view.tree.js]: How about just `download  = enableDownloading`?

**Comment by buunguyen** [comment]
I'm going to close this now. I don't think adding an extra option justifies the slight annoyance of having a download button.

**Comment by Darkle** [comment]
I would really like to have this option. Quite often I accidentally click on the download icon by mistake when I am really trying to click on the file link to display the file as the space in between the icon and the filename will also cause the download to be triggered.

**Comment by jibbers42** [comment]
My use case is a bit different. When looking at angular repos, I don't like the icon changing since angular has very long and similar file names. Often you can't see the extension without making the sidebar very wide. The icons are cool since you don't have to expand the sidebar; however, it's normal to mouse around what you're interested in, but doing so replaces the icon with a download icon :cry:

Ideally, I would prefer the idea at https://github.com/buunguyen/octotree/issues/186#issuecomment-167327341, but browsers make it very easy to right-click download the files, so I like this PR since it's easy and already done.

As a side note @buunguyen, it's a shame you approved the work in #186 and then decided to reject it once someone did it.

**Comment by buunguyen** [comment]
Yeah, sorry about the change of mind. At first I thought it would be a simple change, but looking at the PR, it turns out to be quite complicated. Not to mention, an extra option users have to think about. I decided it's not worth it. One thing most people miss is that while it's easy for me to accept a PR, I am the one who has to maintain the code. Therefore, I need to balance between the value of a PR and the complexity it adds to the codebase.

**Pull request closed without merging**